packages/aws-cdk/lib/notices.ts

```diff
@@ -0,0 +1,270 @@
+    permanentlySuppressed: props.permanentlySuppressed,
+    acknowledgedIssueNumbers: new Set(props.acknowledgedIssueNumbers),
+  }));
+
+  if (individualMessages.length > 0) {
+    cb(finalMessage(individualMessages, notices[0].issueNumber));
+  }
+}
+
+function dataSourceReference(ignoreCache: boolean): NoticeDataSource {
+  return new CachedDataSource(CACHE_FILE_PATH, new WebsiteNoticeDataSource(), ignoreCache);
+}
+
+function finalMessage(individualMessages: string[], exampleNumber: number): string {
+  return [
+    '\nNOTICES',
+    ...individualMessages,
+    `If you don’t want to see a notice anymore, use "cdk acknowledge <id>". For example, "cdk acknowledge ${exampleNumber}".`,
+  ].join('\n\n');
+}
+
+export interface FilterNoticeOptions {
+  outdir?: string,
+  permanentlySuppressed?: boolean,
+  temporarilySuppressed?: boolean,
+  cliVersion?: string,
+  frameworkVersion?: string,
+  acknowledgedIssueNumbers?: Set<number>,
+}
+
+export function filterNotices(notices: Notice[], options: FilterNoticeOptions): Notice[] {
+  const filter = new NoticeFilter({
+    cliVersion: options.cliVersion ?? versionNumber(),
+    frameworkVersion: options.frameworkVersion ?? frameworkVersion(options.outdir ?? 'cdk.out'),
+    temporarilySuppressed: options.temporarilySuppressed ?? false,
+    permanentlySuppressed: options.permanentlySuppressed ?? false,
+    acknowledgedIssueNumbers: options.acknowledgedIssueNumbers ?? new Set(),
+  });
+  return notices.filter(notice => filter.apply(notice));
+}
+
+export function formatNotices(notices: Notice[]): string[] {
+  return notices.map(formatNotice);
+}
+
+export interface Component {
+  name: string;
+  version: string;
+}
+
+export interface Notice {
+  title: string;
+  issueNumber: number;
+  overview: string;
+  components: Component[];
+  schemaVersion: string;
+}
+
+export interface NoticeDataSource {
+  fetch(): Promise<Notice[]>,
+}
+
+export class WebsiteNoticeDataSource implements NoticeDataSource {
+  fetch(): Promise<Notice[]> {
+    return new Promise((resolve) => {
+      https.get('https://cli.cdk.dev-tools.aws.dev/notices.json', res => {
+        if (res.statusCode === 200) {
+          res.setEncoding('utf8');
+          let rawData = '';
+          res.on('data', (chunk) => {
+            rawData += chunk;
+          });
+          res.on('end', () => {
+            try {
+              const notices = JSON.parse(rawData).notices as Notice[];
+              resolve(notices ?? []);
+            } catch (e) {
+              debug(`Failed to parse notices: ${e}`);
+              resolve([]);
+            }
+          });
+          res.on('error', e => {
+            debug(`Failed to fetch notices: ${e}`);
+            resolve([]);
+          });
+        } else {
+          debug(`Failed to fetch notices. Status code: ${res.statusCode}`);
+          resolve([]);
+        }
+      });
+    });
+  }
+}
+
+interface CachedNotices {
+  expiration: number,
```

Comment:
I still find this a little weird. Why wouldn't we use the `mtime` of the cached notices file?

Reply:
Ahhh, you're not wrong. I guess it doesn't matter. I'm having a hard time thinking up cases in which they produce different results.